lib/permutations.rb

```diff
@@ -1,4 +1,31 @@
-
 def permutations?(string1, string2)
-  raise NotImplementedError, "permutations? not implemented" 
+  return true if string1 == string2
+
+  hash = {}
+  hash_2 = {}
+
+  string1.each_char do |letter|
+    hash[letter] = string1.count(letter)
```

Comment:
The `count` is an O(n) operation, so the loop here is O(n^2).

Consider the following

```suggestion
    hash[letter] = 0 if hash[letter].nil?
    hash[letter] += 1
```

Reply:
Thanks! I've submitted with the time/space complexities, please let me know if those are off/wrong.